### PR TITLE
chore(integration): add cases RPC - Miner module

### PR DESCRIPTION
**Pull request by chuijiaolianying**
Add integration test cases for RPC Miner Module:

- get_block_template
- submit_block

### CODE REVIEW

File: test/src/util/epoch_number_with_fraction.rs [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+    epoch.value() & 0xFFFFFF
+}
+
+pub fn get_epoch_index(epoch: EpochNumberWithFraction) -> u64 {
+    (epoch.value() >> 24) & 0xFFFF
+}
+
+pub fn get_epoch_length(epoch: EpochNumberWithFraction) -> u64 {
+    (epoch.value() >> 40) & 0xFFFF
+}
```
Comment (keroro520) [inline test/src/util/epoch_number_with_fraction.rs:16]: Just a suggestion: add the corresponding implementation to `ckb-jsonrpc-types::EpochNumberWithFraction`, of cause, in another PR.

File: test/src/specs/rpc/submit_block.rs [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+            .unwrap_err();
+        assert!(
+            block_err.to_string().contains("UnknownParentError"),
+            "Submit block with wrong parent hash should return 'UnknownParentError'"
+        );
+
+        // build block with wrong timestamp: block time too new
+        let block = node0
+            .new_block_builder(None, None, None)
+            .timestamp((now_ms() + 16_000).pack())
```
Comment (keroro520) [inline test/src/specs/rpc/submit_block.rs:92]: Add some comments around `16_000` and the relation with `ALLOWED_FUTURE_BLOCKTIME`.
And suggest `30s` as the ci machine is poor.

**Review by keroro520 — APPROVED** [review]

File: test/src/util/epoch_number_with_fraction.rs (hunk 1 above)
Comment (chuijiaolianying) [reply test/src/util/epoch_number_with_fraction.rs:16]: Thanks for your suggestion, create a new PR for this: https://github.com/nervosnetwork/ckb/pull/2495

File: test/src/specs/rpc/submit_block.rs (hunk 2 above)
Comment (chuijiaolianying) [reply test/src/specs/rpc/submit_block.rs:92]: I'm going to add comment about this together with updating epoch relate functions after #2495

File: test/src/util/epoch_number_with_fraction.rs (hunk 1 above)
Comment (quake) [reply test/src/util/epoch_number_with_fraction.rs:16]: Please rebase when #2495 is merged

**Comment by doitian** [comment]
#2495 has been merged. Waiting for author to apply the review suggestions above.

**Comment by chuijiaolianying** [comment]
@doitian  @keroro520  PTAL, thx

**Review by doitian — APPROVED** [review]

**Review by quake — APPROVED** [review]

**Comment by zhangsoledad** [comment]
bors r=doitian,quake

**Comment by doitian** [comment]
bors retry

**Pull request merged**